src/index.js

```diff
@@ -135,13 +135,22 @@ function matchesSelector (element, ast) {
       if (node.value === ' ') {
         // walk all ancestors
         const precedingNodes = getLastNonCombinatorNodes(nodes.slice(0, i))
-        const ancestor = getFirstMatchingAncestor(element, precedingNodes)
+        const prePrecedingNodes = nodes.slice(0, i - precedingNodes.length)
+        let ancestor = getFirstMatchingAncestor(element, precedingNodes)
         if (!ancestor) {
           return false
-        } else {
-          element = ancestor
-          i -= precedingNodes.length
         }
+        // Even though first ancestor matches the selector, ancestor should match all preceding nodes
+        while (ancestor) {
+          // If this ancestor is compatible with the preceding nodes, then it is a match
+          // If not, walk up the ancestor tree until a match is found
+          if (matchesSelector(ancestor, { nodes: prePrecedingNodes })) {
+            return true
+          }
+          ancestor = getFirstMatchingAncestor(ancestor, precedingNodes)
+        }
+        // While loop has exhausted all the possible ancestors and not found a match
+        return false
```

Comment:
It seems incorrect to me that we are only doing this logic when `node.value === ' '`. We should also handle `node.value === '>'`.

For example, this selector should also match (note the extra `>`):

```diff
- outer-component > div.container > div:nth-child(1) .selectBox1
+ outer-component > div.container > div:nth-child(1) > .selectBox1
```

Reply:
Nevermind – I realize now why we don't need to do the extra loop for `node.value === '>'`. In that case, we only need to match against the parent, and there's only one of those.